devices/debian.py

```diff
@@ -752,16 +752,19 @@ def link_up(self, interface):
         self.sendline("ip link show %s" % interface)
         self.expect(self.prompt)
         link_state = self.before
-        match = re.search('NO-CARRIER,BROADCAST,MULTICAST,UP',link_state)
+        match = re.search('BROADCAST,MULTICAST,UP',link_state)
         if match:
             return match.group(0)
         else:
             return None
 
     def set_link_state(self, interface, state):
         '''Setting the interface status'''
-        self.sendline("ip link set %s %s" % (interface,state))
-        self.expect(self.prompt)
+        self.sudo_sendline("ip link set %s %s" % (interface,state))
+        idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
```

Comment:
This logic should go INSIDE sudo_sendline() not be repeated everywhere...

Reply:
Ack